### PR TITLE
Configure and document goreleaser for the provider

### PR DESCRIPTION
**Pull request by ziemowit141**
https://clkentik.atlassian.net/browse/KNTK-250

### CODE REVIEW

File: .goreleaser.yml [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+archives:
+- format: zip
+  name_template: '{{ .ProjectName }}_{{ .Version }}_{{ .Os }}_{{ .Arch }}'
+checksum:
+  name_template: '{{ .ProjectName }}_{{ .Version }}_SHA256SUMS'
+  algorithm: sha256
+signs:
+  - artifacts: checksum
+    args:
+      # if you are using this in a GitHub action or some other automated pipeline, you
```
Comment (Fenthick) [inline .goreleaser.yml:41]:
```suggestion
```

File: .goreleaser.yml [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+- format: zip
+  name_template: '{{ .ProjectName }}_{{ .Version }}_{{ .Os }}_{{ .Arch }}'
+checksum:
+  name_template: '{{ .ProjectName }}_{{ .Version }}_SHA256SUMS'
+  algorithm: sha256
+signs:
+  - artifacts: checksum
+    args:
+      # if you are using this in a GitHub action or some other automated pipeline, you 
+      # need to pass the batch flag to indicate its not interactive.
```
Comment (Fenthick) [inline .goreleaser.yml:42]:
```suggestion
```

File: .github/workflows/release.yml [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+jobs:
+  goreleaser:
+    runs-on: ubuntu-latest
+    steps:
+      -
+        name: Checkout
+        uses: actions/checkout@v2.3.4
+      -
+        name: Unshallow
+        run: git fetch --prune --unshallow
```
Comment (Fenthick) [inline .github/workflows/release.yml:26]: Just pass `fetch_depth: 0` to checkout action and remove this step.
See: https://github.com/marketplace/actions/checkout

**Review by Fenthick — CHANGES_REQUESTED** [review]
Mostly stylistic comments

File: README.md [hunk 4]
```diff
@@ -1,5 +1,14 @@
 # Terraform Provider for Kentik synthetic tests
 
+## Release process
```
[the remark this answers is not in the document]
Comment (ziemowit141) [reply README.md]: Addressed here: 7e7c7fc8707dd7b4d7d50a802ada8bcc9ef70a63

File: .goreleaser.yml (hunk 2 above)
Comment (ziemowit141) [reply .goreleaser.yml]: Addressed here: eb3464766644488c4b302fde1aaad0b790981e86

File: .goreleaser.yml (hunk 1 above)
Comment (ziemowit141) [reply .goreleaser.yml]: Addressed here: eb3464766644488c4b302fde1aaad0b790981e86

File: .goreleaser.yml [hunk 5]
```diff
@@ -0,0 +1,54 @@
+# Visit https://goreleaser.com for documentation on how to customize this behavior.
+before:
+  hooks:
+    - go mod tidy
```
[the remark this answers is not in the document]
Comment (ziemowit141) [reply .goreleaser.yml]: Addressed here: 062717dd1c35dc7571fa9e00276c49d9ee886292

File: .github/workflows/release.yml (hunk 3 above)
Comment (ziemowit141) [reply .github/workflows/release.yml]: Addressed here: ce694377e56cabf8b78dfb2ae4f6295d04de9877

**Review by Fenthick — APPROVED** [review]

**Comment by Fenthick** [comment]
@mmac-m3a Please review/merge

**Review by mmac-m3a — APPROVED** [review]
LGTM

**Pull request merged**